Rename "user stories" to "feature concepts" (#115)
simplabs/playbook: src/project-procedure/digital-products/README.md

using the real application to collect feedback and validate assumptions as early
as possible.

### Feature Concepts Workshop

> In the [Feature Concepts](https://en.wikipedia.org/wiki/User_story) Workshop, the

Reviewer: Can you remove the link? It is no longer correct.

This workshop requires input from all project stakeholders:

- The business experts have unique expertise and understand best what goal a

>   particular feature concept enables the user to achieve

Reviewer: ```suggestion
  particular feature enables the user to achieve
```

- Designers have the expertise to understand which interaction design lets the
  user achieve the above goal in the most efficient way

> - Engineers can assess different alternatives for implementing a feature concept for

Reviewer: ```suggestion
- Engineers can assess different alternatives for implementing a feature for
```

  their technical feasibility and associated effort

> Each coherent feature of the application is represented by a feature concept. Feature concepts enable one or more personas (as identified during the

Reviewer: ```suggestion
Each coherent feature of the application is represented by a feature concept. Features enable one or more personas (as identified during the
```

[product strategy stage](#product-strategy)) to perform a particular task in
order to achieve a particular goal.

For each feature concept, identify:

- The navigational structure: screens or pages, dialogues, menus etc.
- Interactive elements: buttons, inputs etc. that the user needs to trigger
  actions in the flow or needs to acquire information that is relevant to them
- Flows: transitions between sections of the navigational structure that are
  triggered by interactive elements

Feature concepts are visualized using coarse diagrams (e.g. hand-drawn marker
sketches) including all of these identified pieces and the relationships between
them. During the workshop participants will discuss and reconstruct these
diagrams several times until an ideal solution emerges that all stakeholders
agree on. That final concept is then written down in a feature concept document
consisting of:

> - A textual description of the feature concept, the need that it addresses and the

Reviewer: ```suggestion
- A textual description of the feature, the need that it addresses and the
```

  solution that it provides

> - The persona(s) that the feature concept (mainly) caters to

Reviewer: ```suggestion
- The persona(s) that the feature (mainly) caters to
```

> - The low-fidelity diagram showing steps in the feature concept, focusing on

Reviewer: ```suggestion
- The low-fidelity diagram showing steps in the feature, focusing on
```

  functional aspects and transitions between steps

![Feature concept](../../../assets/feature-concept.svg "Feature Concepts are defined focussing on the flow and relevant interactive elements rather than concrete visuals")

> Once a feature concept has been clearly defined, revisit it and look for

Reviewer: ```suggestion
Once a feature concept has been clearly described, revisit it and look for
```

non-essential aspects that are not strictly necessary in order for the feature

> concept to provide value as well as edge cases that might not immediately be

Reviewer: ```suggestion
to provide value as well as edge cases that might not immediately be
```

apparent. These will be noted in the feature concept document as well and can be used

> as escape hatches later on to limit the scope and effort for the feature concept as

Reviewer: ```suggestion
as escape hatches later on to limit the scope and effort for the feature as
```

up-front to create brand guidelines (if none exist) and establish variables such
as typography, color, and spacing.

> Next, create mid-fidelity wireframes of the first feature concept to convert the

Reviewer: ```suggestion
Next, create mid-fidelity wireframes of the first feature or coherent set of features to convert the
```

functionality to concrete UI elements, but without potentially distracting
stylistic details of high-fidelity designs. Once the wireframes are agreed upon

> by all relevant stakeholders, mockups of the feature concept can be created. Use the

Reviewer: ```suggestion
by all relevant stakeholders, mockups of the feature can be created. Use the
```

ideas. Their insights informs the design, helping evolve it iteratively
according to their actual needs.

> ![Wireframe and Mockup](../../../assets/wireframe-and-mockup.svg "The first increment of a product will require creating wireframes and mockups for one of the feature concepts in order to agree on a layout and visual style for the product")

Reviewer: ```suggestion
![Wireframe and Mockup](../../../assets/wireframe-and-mockup.svg "The first increment of a product will require creating wireframes and mockups for one of the features in order to agree on a layout and visual style for the product")
```

### Increment Planning

> Once the feature concepts are defined in the product design stage, define a plan for

Reviewer: ```suggestion
Once the feature concepts are described in the product design stage, define a plan for
```

> the current increment of the application. For each feature concept that is part of an

Reviewer: ```suggestion
the current increment of the application. For each feature that is part of an
```

increment, list which of the non-essential aspects and edge cases will be
included in the increment (e.g. configuring products of the most successful
category), which ones will not (e.g. configuring products with little sales
volume) and which ones will be moved to a later increment in case the team is
running out of time or budget (e.g. allowing customers to see the status of past
orders). Introducing these escape hatches allows the project team to make a

> decision between implementing essential versions of more feature concepts versus

Reviewer: ```suggestion
decision between implementing essential versions of more features versus
```

> covering more non-essential aspects and edge cases for fewer feature concepts later

Reviewer: ```suggestion
covering more non-essential aspects and edge cases for fewer features later
```

in the process if time should become scarce.

Once all stakeholders agree on the plan for the product increment, the list of
feature concepts together with included as well as excluded aspects and according
edge cases are written down in an increment plan document, which is used to
estimate a rough timeline. That plan is not irrevocable and neither is the
estimate more than an assessment based on the information available at the time.
Previously unknown challenges will likely be uncovered during product
development and some details and implications will only be understood once work

> on a particular feature concept has actually begun. Teams should in fact constantly

Reviewer: ```suggestion
on a particular feature has actually begun. Teams should in fact constantly
```

question assumptions and priorities and change the plan at any time they see
necessary. It is also possible at any time to go back and re-plan an increment,

> adding a different set of feature concepts, include or exclude different aspects of

Reviewer: ```suggestion
adding a different set of features, include or exclude different aspects of
```

## Product Development

In the product development stage of a product increment, feature concepts are turned
into usable application features, iteratively building up the product over time.
During this process, each feature concept is broken down into fine-grained and more
detailed tasks which are then proceeded to be implemented by the project team
(see [development process](../development-process/#development-process)).
Breaking down feature concepts is done in close collaboration between engineers and

> designers with the business experts. They discuss individual facets of feature concepts

Reviewer: ```suggestion
designers with the business experts. They discuss individual facets of features
```

as well as different alternatives for designing and implementing those
along with the associated effort.

The project team will build slices of the system at once, including everything

> from the design, the backend, and the frontend code of a particular feature concept

Reviewer: ```suggestion
from the design, the backend, and the frontend code of a particular feature
```

### Growing a Foundation

> As more feature concepts are being completed and the product evolves, foundational

Reviewer: ```suggestion
As more feature are being completed and the product evolves, foundational
```

elements that will later be reused in other, similar contexts will be
identified - both in the application's codebase as well as in the
[visual design system](../../development-process/#design-systems). That way,
each increment of the application results in its foundation becoming broader,
stronger and more refined, resulting in increased effectiveness over time,

> decreasing effort per feature concept and increasing planning reliability.

Reviewer: ```suggestion
decreasing effort per feature and increasing planning reliability.
```
